sql/catalyst/src/test/scala/org/apache/spark/sql/util/TimestampFormatterSuite.scala

```diff
@@ -64,7 +65,8 @@ class TimestampFormatterSuite extends SparkFunSuite with SQLHelper with Matchers
     DateTimeTestUtils.outstandingTimezonesIds.foreach { zoneId =>
       val formatter = TimestampFormatter(
         "yyyy-MM-dd'T'HH:mm:ss.SSSSSS",
-        DateTimeUtils.getZoneId(zoneId))
+        DateTimeUtils.getZoneId(zoneId),
+        needVarLengthSecondFraction = true)
```

Comment:
ditto

Reply:
this one is not needed

Reply:
really? this test is for formatting, and the actual formatting doesn't do padding, so this flag should be false.

Reply:
oh. I mean this should be false too.